File: php/backend/App_service/app/Services/AuthService.php
<?php

namespace App\Services;

use App\DTOs\LoginDTO;
use App\DTOs\SignupDTO;
use App\Models\User;
use App\Repositories\UserRepository;
use Illuminate\Support\Facades\Hash;

class AuthService
{
    public function __construct(private UserRepository $users) {}

    public function signup(SignupDTO $dto): array
    {
        if ($this->users->emailExists($dto->email)) {
            return ['success' => false, 'message' => 'User already exists'];
        }

        if (strlen($dto->password) < 10) {
        return [
            'success' => false,
            'message' => 'Password is too weak. It must be at least 10 characters. Example: MyPass@2024'
            ];
        }

        $user = $this->users->create([
            'name'     => $dto->name,
            'email'    => $dto->email,
            'phone'    => $dto->phone,
            'password' => Hash::make($dto->password),
        ]);

        return ['success' => true, 'user' => $user];
    }

    public function login(LoginDTO $dto): array
    {
        $user = $this->users->findByEmail($dto->email);

        if (!$user || !Hash::check($dto->password, $user->password)) {
            return ['success' => false, 'message' => 'Invalid credentials'];
        }

        $token = bin2hex(random_bytes(32));
        return ['success' => true, 'token' => $token, 'user' => $user];
    }
}

File: csharp/backend/SERVICE/Authservice.cs
using LeadApi.Models;

namespace LeadApi.Services;

public class AuthService(PostgresStore store)
{
    public async Task<(string Token, AppUser User)?> LoginAsync(LoginRequest request)
    {
        var user = await store.LoginAsync(request);
        if (user is null) return null;
        return (Guid.NewGuid().ToString(), user);
    }

    public async Task<(bool Success, string Message, AppUser? User)> SignupAsync(SignupRequest request)
    {

        if (await store.EmailExistsAsync(request.Email))
        {
            return (false, "User already exists", null);
        }

        if (request.Password.Length < 6)
        {
            return (false, "Password must be at least 6 characters long", null);
        }

        var user = new AppUser
        {
            Id= Guid.NewGuid().ToString(),
            Name= request.Name,
            Email= request.Email,
            Phone= request.Phone,
            CreatedAt= DateTimeOffset.UtcNow
        };
        var savedUser = await store.SaveUserAsync(user, request.Password);
        return (true, "Signup successfully", savedUser);
    }
}


// using LeadApi.Models;

// namespace LeadApi.Services;

// public class AuthService(InMemoryStore store)
// {
//     public (string Token, AppUser User)? Login(LoginRequest request)
//     {
//         var user = store.Login(request);
//         if (user is null) return null;

//         var token = Guid.NewGuid().ToString(); // swap for JWT in production
//         return (token, user);
//     }

//     public AppUser Signup(SignupRequest request)
//     {
//         var user = new AppUser(
//             Id: Guid.NewGuid().ToString(),
//             Name: request.Name,
//             Email: request.Email,
//             Phone: request.Phone,
//             CreatedAt: DateTimeOffset.UtcNow
//         );
//         return store.SaveUser(user, request.Password);
//     }
// }